src/Repo_old.java

```diff
@@ -0,0 +1,804 @@
+                        }
+                    }
+                });
+
+                if (!approved[0]) {
+                    throw new ConfigurationException("Failed to login to "
+                            + host + ':' + port);
+                }
+            } else {
+                if (userName == null) {
+                    userName = props.getOrRequestProperty(PROP_USER,
+                            "Username: ", false);
+                }
+                if (password == null) {
+                    password = props.getOrRequestProperty(PROP_PASSWORD,
+                            "Password: ", true);
+                }
+                if (server.logOn(userName, password) == 0) {
+                    throw new ConfigurationException("Failed to login to "
+                            + host + ':' + port + " as user " + userName);
+                }
+                cacheLogOnCredentials(server, userName, password);
+            }
+        }
+
+        return server;
+    }
+
+    public synchronized View connect() {
+        if (_view == null) {
+            final RepoProperties props = getRepoProperties();
+            final StarTeamURL url = getUrl();
+            final String project = url.getProjectName();
+            final String view = getViewName(url);
+            final Server server = getIdleConnection();
+            boolean ok = false;
+
+            try {
+                if (getLogger().isDebugEnabled()) {
+                    final java.io.File dir = new File(props.getGitstDir(),
+                            getBranchName());
+                    dir.mkdirs();
+                    NetMonitor.onFile(new java.io.File(dir, "NetMonitor.log"));
+                }
+
+                if (getLogger().isInfoEnabled()) {
+                    final int port = Integer.parseInt(url.getPort());
+                    final String host = url.getHostName();
+                    getLogger().info(
+                            "Connecting to " + host + ':' + port + '/'
+                                    + project + '/' + view);
+                }
+
+                _view = findView(findProject(server, project), view);
+                _rootFolder = getRootFolder(url, _view);
+                ok = true;
+            } finally {
+                if (ok) {
+                    releaseConnection(server);
+                } else {
+                    close();
+                    _pool.clear();
+                }
+            }
+        }
+
+        return _view;
+    }
+
+    @Override
+    public synchronized void close() {
+        if (_view != null) {
+            close(_view);
+
+            for (final View v : _viewCache) {
+                close(v);
+            }
+
+            _view = null;
+            _rootFolder = null;
+            _pool.clear();
+            _viewCache.clear();
+        }
+    }
+
+    private void close(final View v) {
+        try {
+            v.close();
+        } catch (final Throwable ex) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1181)